### PR TITLE
Add IPython notebooks for historical data sets T006–T010, S012

### PR DESCRIPTION
**Pull request by hlinero**
**Note:** this PR is retained for reference. The notebooks will not be merged, but converted to Python submodules, in the manner of #31.

---
The purpose of this pull request is to merge the scripts T007 to S012 into the master branch.

### CODE REVIEW

**Comment by khaeru** [comment]
Added the assignee, label, and milestone for this PR.

File: item/historical/scripts/DatasetMerger.ipynb [hunk 1]
```diff
@@ -2,7 +2,7 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": 27,
+   "execution_count": 1,
```
Comment (khaeru) [inline item/historical/scripts/DatasetMerger.ipynb]: Please revert these changes to cell execution counts. They don't constitute meaningful changes to the _functionality_ of the code, so having them in the history makes it difficult to understand the actual changes.

File: item/historical/scripts/DatasetMerger.ipynb [hunk 2]
```diff
@@ -191,147 +185,141 @@
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
-       "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
-       "      <th>82553</th>\n",
```
Comment (khaeru) [inline item/historical/scripts/DatasetMerger.ipynb]: Similarly, if we insist on having the data displayed at the top of every notebook, we are going to have thousands of diff lines like this to wade through, every time the data changes even slightly.

I would suggest removing the output altogether. We can merge this PR without removing the output, but need to find a solution for the next PR.

File: item/historical/scripts/DatasetMerger.ipynb [hunk 3]
```diff
@@ -346,6 +334,11 @@
     "list_of_paths.append(\"/Users/hlinero/Desktop/iTEM Material/raw dataset/T004/T004_cleaned_PF.csv\")\n",
     "list_of_paths.append(\"/Users/hlinero/Desktop/iTEM Material/raw dataset/T005/T005_cleaned_PF.csv\")\n",
     "list_of_paths.append(\"/Users/hlinero/Desktop/iTEM Material/raw dataset/T006/T006_cleaned_PF.csv\")\n",
+    "list_of_paths.append(\"/Users/hlinero/Desktop/iTEM Material/raw dataset/T007/T007_cleaned_PF.csv\")\n",
```
Comment (khaeru) [inline item/historical/scripts/DatasetMerger.ipynb]: Remove hard-coded paths that are specific to one user's machine.

File: item/historical/scripts/S012.ipynb [hunk 4]
```diff
@@ -0,0 +1,3078 @@
...
+    "# Variables used across the noteboook and not changed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATASET_ID = \"S012\"\n",
```
Comment (khaeru) [inline item/historical/scripts/S012.ipynb]: Why is this script called 'S...' instead of 'T...'? The [documentation](https://transportenergy.readthedocs.io/en/latest/historical.html) should provide enough information for a newcomer to understand this nomenclature and what the scripts contain, _without_ having to open the individual notebooks.

File: item/historical/scripts/util/managers/dataframe.py [hunk 5]
```diff
@@ -160,3 +160,34 @@ def create_user_friendly_file(self, df):
         path = "{}/{}".format(cwd, filename)
         df_with_all_countries_data.to_csv(path, index=False)
         print("> UF File saved at: {}".format(cwd))
+
+    def transform_from_uf_to_pf_view(self, df, num_of_none_numeric_columns):
+        '''
```
Comment (khaeru) [inline item/historical/scripts/util/managers/dataframe.py]: Please use [numpydoc format docstrings](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard), as indicated in our [style guide](https://transportenergy.readthedocs.io/en/latest/reference.html#style-guide).

File: item/historical/scripts/T000.ipynb [hunk 6]
```diff
@@ -364,9 +364,667 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 4,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
```
Comment (khaeru) [inline item/historical/scripts/T000.ipynb]: The description of this PR only mentions "merging scripts T007 to S012". What are these changes to T000? We need to have an intelligible description of all changes in a PR, in order to:
- not force reviewers to read the code line-by-line and guess at the intent and effect of changes, and
- allow others, later, to understand why those changes were made.

File: item/historical/scripts/T001.ipynb [hunk 7]
```diff
@@ -562,7 +562,7 @@
      "text": [
       ">> Number of countries missing values: 16\n",
       ">> Countries missing values:\n",
-      "['Greece', 'Hungary', 'Slovak Republic', 'Switzerland', 'Serbia, Republic of', 'Czech Republic', 'Liechtenstein', 'North Macedonia', 'Belarus', 'Austria', 'Montenegro, Republic of', 'Moldova', 'Armenia', 'Slovenia', 'Luxembourg', 'Lithuania']\n",
+      "['Liechtenstein', 'North Macedonia', 'Slovenia', 'Czech Republic', 'Slovak Republic', 'Lithuania', 'Luxembourg', 'Austria', 'Moldova', 'Hungary', 'Belarus', 'Armenia', 'Switzerland', 'Serbia, Republic of', 'Greece', 'Montenegro, Republic of']\n",
```
Comment (khaeru) [inline item/historical/scripts/T001.ipynb]: Ditto comment on T000. Is this a meaningful change, or just an unsorted list that has changed order?

File: item/historical/scripts/T002.ipynb [hunk 8]
```diff
@@ -561,21 +561,21 @@
      "text": [
       ">> Number of countries missing values: 15\n",
       ">> Countries missing values:\n",
-      "['Malta',\n",
-      " 'Austria',\n",
+      "['Iceland',\n",
```
Comment (khaeru) [inline item/historical/scripts/T002.ipynb]: Ditto comment on T000.

File: item/historical/scripts/T003.ipynb [hunk 9]
```diff
@@ -561,35 +561,35 @@
      "text": [
       ">> Number of countries missing values: 29\n",
       ">> Countries missing values:\n",
-      "['Korea',\n",
-      " 'North Macedonia',\n",
+      "['Ireland',\n",
```
Comment (khaeru) [inline item/historical/scripts/T003.ipynb]: Ditto comment on T000.

File: item/historical/scripts/T004.ipynb [hunk 10]
```diff
@@ -2,7 +2,7 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": 22,
```
Comment (khaeru) [inline item/historical/scripts/T004.ipynb]: This file appears to only have changes to execution counts. See comment on DatasetMerger.ipynb.

File: item/historical/scripts/T005.ipynb [hunk 11]
```diff
@@ -2170,29 +2170,29 @@
...
-   "execution_count": 36,
+   "execution_count": 8,
    "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "['Cape Verde',\n",
-       " 'Virgin Islands_British',\n",
+       "['Serbia and Montenegro',\n",
```
Comment (khaeru) [inline item/historical/scripts/T005.ipynb]: Ditto comment on T000.

File: item/historical/scripts/T006.ipynb [hunk 12]
```diff
@@ -1009,6 +1009,185 @@
     "df"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Renaming the \"Germany\" country name\n",
```
Comment (khaeru) [inline item/historical/scripts/T006.ipynb]: Ditto comment on T000.

File: item/historical/scripts/T007.ipynb [hunk 13]
```diff
@@ -1,5 +1,3236 @@
...
+   "metadata": {},
+   "source": [
+    "# Opening the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
```
Comment (khaeru) [inline item/historical/scripts/T007.ipynb]: Please clear outputs for these newly-added notebooks to avoid the problems mentioned above.

File: item/historical/scripts/DatasetMerger.ipynb (hunk 3 above)
Comment (hlinero) [reply item/historical/scripts/DatasetMerger.ipynb]: This file is going to change, possibly be erased soon, but I still need it for now. So, we can either ignore it in the gitignore or just wait a bit until I change it.

**Comment by khaeru** [comment]
Superseded by #71.
